kserve builds on master on x86_64-linux but breaks after this PR

```
  > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED test/test_server.py::TestTFHttpServerLoadAndUnLoad::test_unload - assert <HTTPStatus.NOT_FOUND: 404> == 200
       > ============ 1 failed, 315 passed, 20 warnings in 88.38s (0:01:28) =============
       For full logs, run 'nix log /nix/store/1fkrcg3bv1q3fjmbmfh5jfdl0h5lygbh-python3.12-kserve-0.14.1.drv'.
error: 1 dependencies of derivation '/nix/store/p8l1bfpgl71vfgf7gvg1wjxgzg61h2i3-review-shell.drv' failed to build
```

--- a/pkgs/development/python-modules/kserve/default.nix
+++ b/pkgs/development/python-modules/kserve/default.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
 
@@ -121,19 +122,44 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "kserve" ];
 
+  pytestFlagsArray =
+    [
+      # AssertionError
+      "--deselect=test/test_server.py::TestTFHttpServerLoadAndUnLoad::test_unload"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # RuntimeError: Failed to start GCS
+      "--deselect=test/test_dataplane.py::TestDataPlane::test_explain"
+      "--deselect=test/test_dataplane.py::TestDataPlane::test_infer"
+      "--deselect=test/test_dataplane.py::TestDataPlane::test_model_metadata"
+      "--deselect=test/test_dataplane.py::TestDataPlane::test_server_readiness"
+      "--deselect=test/test_server.py::TestRayServer::test_explain"
+      "--deselect=test/test_server.py::TestRayServer::test_health_handler"
+      "--deselect=test/test_server.py::TestRayServer::test_infer"
+      "--deselect=test/test_server.py::TestRayServer::test_list_handler"
+      "--deselect=test/test_server.py::TestRayServer::test_liveness_handler"
+      "--deselect=test/test_server.py::TestRayServer::test_predict"
+    ];
+
   disabledTestPaths = [
     # Looks for a config file at the root of the repository
     "test/test_inference_service_client.py"
   ];
 
-  disabledTests = [
-    # Require network access
-    "test_infer_graph_endpoint"
-    "test_infer_path_based_routing"
+  disabledTests =
+    [
+      # Require network access
+      "test_infer_graph_endpoint"
+      "test_infer_path_based_routing"
 
-    # Tries to access `/tmp` (hardcoded)
-    "test_local_path_with_out_dir_exist"
-  ];
+      # Tries to access `/tmp` (hardcoded)
+      "test_local_path_with_out_dir_exist"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      "test_local_path_with_out_dir_not_exist"
+    ];
+
+  __darwinAllowLocalNetworking = true;
 
   meta = {
     description = "Standardized Serverless ML Inference Platform on Kubernetes";

--- a/pkgs/development/python-modules/kserve/default.nix
+++ b/pkgs/development/python-modules/kserve/default.nix
@@ -44,6 +44,7 @@
   avro,
   grpcio-testing,
   pytest-asyncio,
+  pytest-xdist,
   pytestCheckHook,
   tomlkit,
 }:
@@ -116,6 +117,7 @@ buildPythonPackage rec {
     avro
     grpcio-testing
     pytest-asyncio
+    pytest-xdist
     pytestCheckHook
     tomlkit
   ] ++ lib.flatten (builtins.attrValues optional-dependencies);

